packages/teams-js/src/internal/communication.ts

```diff
@@ -605,6 +614,65 @@ function updateRelationships(messageSource: Window, messageOrigin: string): void
 
 const handleParentMessageLogger = communicationLogger.extend('handleParentMessage');
 
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+function retrieveMessageUUIDFromResponse(response: MessageResponse): MessageUUID {
+  if (response.uuid) {
+    const responseUUID = response.uuid;
+    const callbackUUID = retrieveMessageUUIDFromCallback(CommunicationPrivate.callbacks, responseUUID);
+    if (callbackUUID) {
+      return callbackUUID;
+    }
+    const promiseCallbackUUID = retrieveMessageUUIDFromCallback(CommunicationPrivate.promiseCallbacks, responseUUID);
+    if (promiseCallbackUUID) {
+      return promiseCallbackUUID;
+    }
+    const portCallbackUUID = retrieveMessageUUIDFromCallback(CommunicationPrivate.portCallbacks, responseUUID);
+    if (portCallbackUUID) {
+      return portCallbackUUID;
+    }
+  }
+  return CommunicationPrivate.legacyMessageIdsToUuidMap[response.id];
+}
+
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * This function is used to compare a new MessageUUID object value to the key values in the specified callback and retrieving that key
+ * We use this because two objects with the same value are not considered equivalent therefore we can't use the new MessageUUID object
+ * as a key to retrieve the value assosciated with it and should use this function instead.
+ */
+function retrieveMessageUUIDFromCallback(
+  map: Map<MessageUUID, Function>,
+  responseUUID: MessageUUID,
+): MessageUUID | undefined {
+  const callback = [...map].find(([key, _value]) => {
+    return key.toString() === responseUUID.toString();
+  });
+
+  if (callback) {
+    return callback[0];
+  }
+  return undefined;
+}
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+function removeMessageHandlers(
+  message: MessageResponse,
+  callbackId: MessageUUID,
+  map: Map<MessageUUID, unknown>,
+): void {
+  map.delete(callbackId);
```

Comment:
Since special handling is needed to "find" a uuid in the map (E.g. that function above here that compares their string values), would similar special handling be needed here? Otherwise how could it reliably find which entry in the map to delete?